src/components/Navbar.js

```diff
@@ -78,22 +156,30 @@ class BounceNavbar extends Component {
                     <Navbar.Toggle />
                 </Navbar.Header>
                 <Navbar.Collapse>
-                    <Navbar.Form>
-                        <form onSubmit={this.handleSubmit}>
-                            <FormGroup>
-                                <FormControl
-                                    type='text'
-                                    placeholder='Search'
-                                    onChange={this.handleInput}
-                                />
-                            </FormGroup>
-                            <Button type='submit'>
-                                <Glyphicon glyph='search'></Glyphicon>
-                            </Button>
-                        </form>
-                    </Navbar.Form>
+                    <Nav>
+                        <NavItem eventKey={1} href="#">
+                            <Navbar.Form>
+                                <form onSubmit={this.handleSubmit}>
+                                    <FormGroup>
+                                        <FormControl
+                                            type='text'
+                                            placeholder='Search'
+                                            onChange={this.handleInput}
+                                        />
+                                    </FormGroup>
+                                    <Button type='submit'>
+                                        <Glyphicon glyph='search'></Glyphicon>
+                                    </Button>
+                                </form>
+                            </Navbar.Form>
+                        </NavItem>
+                    </Nav>
+                    
+                    {navbarComponent}
                 </Navbar.Collapse>
-                {homeRedirect}
+                
+                {pageRedirect}
```

Comment:
Minor, I think you can just put the page redirect at the top, since it'll trigger the rendering of a different page anyway.